smoke-test/smoke-test.sh

```diff
@@ -59,22 +59,14 @@ node generated/tsc/smoke-test.mjs
 # because we used to really like putting `<` engine constraints on all our
 # packages.
 #
-# `--node-version=16.0.0` tricks npm into thinking we are Node v16 for the
-# purpose of choosing versions. Even with `--no-engine-strict` or `--force`,
-# when it evaluates `@apollo/gateway@0.x` it will rule out versions whose
-# `engines` rule out the current Node version. We should hopefully publish a new
-# latest 0.x version soon which doesn't have `< 18` for Node but for now lie
-# about our version. (This flag will go away in npm v9 so this is only a
-# short-term solution.)
-#
 # This runs into some weird issues when we install the graphql@17 canary that
 # seems to just be about ending up with two copies of `graphql` installed at
 # different levels. For simplicity, we skip this part of the test in the canary
 # job. Once graphql@17.0.0 is out we should be able to de-conditional this.
 if [[ -z "${INCREMENTAL_DELIVERY_TESTS_ENABLED:-}" ]]; then
   pushd gateway-compatibility
     for version in 0.50.1 0.51.0 0.x 2.0.0 2.0.5 2.x; do
```

Comment:
@glasser Seems like we'd need to drop the 0.50.x's for this to work so I'm not sure why that didn't break, am I missing something?

Reply:
We are still passing `--no-engine-strict` so `npm i` lets us do the installation.

The issue before was that the `engines` was also affecting *choosing which version to use within a range*, and so `0.x` was evaluating to some ultra ultra old version from before we had added the `<18` at all.  Surprisingly, `--no-engine-strict` did not affect that logic and I couldn't find any other way to turn it off other than lying about the version. But now we have a reasonable engine-compatible 0x choice.

This didn't affect 0.50.0 or 0.50.1 because there is *no* "better" choice matching those "ranges".